build_probe.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -eux
+
+JAVA_MAJOR_VERSION=$1
+BASE_DOCKER_IMAGE="openjdk:${JAVA_MAJOR_VERSION}-jre-slim"
+
+docker pull "${BASE_DOCKER_IMAGE}"
+
+EXACT_JAVA_VERSION=$(docker run --rm "$BASE_DOCKER_IMAGE" java -version 2>&1 | awk -F '"' '/version/ {print $2}')
+
+docker build \
+  --build-arg=JAVA_VERSION="$JAVA_MAJOR_VERSION" \
+  -t gatlingcorp/frontline-injector:"$EXACT_JAVA_VERSION" \
+  .
+
+docker tag \
+  gatlingcorp/frontline-injector:"$EXACT_JAVA_VERSION" \
+  gatlingcorp/frontline-injector:"$JAVA_MAJOR_VERSION"
+
+docker push gatlingcorp/frontline-injector:"$EXACT_JAVA_VERSION"
```

Comment:
We shouldn't publish public images with exact Java version. Otherwise, we won't be able to delete them as they might be used by our customers and we'll end up with a huge pile of junk.

Reply:
I do not agree with that analysis:

* Publishing images with exact versions allows to rollback a probe update by removing the faulty image and alias a previous image with a specific version to the Java major version.
* The Kubernetes pools are configured to always pull the latest version of the certified image, meaning that users will get automatic updates and the old images will stop being used.
* Users who references directly a specific version can and should copy the image (same as someone who'd reference directly a certified AMI by its ID rather than letting FrontLine fetch the latest)
* As much as possible, we should consider the Docker Hub as immutable, it's another repository of releases, and we should avoid overwriting/deleting images on it.